src/pymor/vectorarrays/block.py

```diff
@@ -7,116 +7,107 @@
 import numpy as np
 
 from pymor.core.base import classinstancemethod
-from pymor.vectorarrays.interface import VectorArray, VectorSpace
+from pymor.tools.deprecated import Deprecated
+from pymor.vectorarrays.interface import VectorArray, VectorArrayImpl, VectorSpace
 
 
-class BlockVectorArray(VectorArray):
-    """|VectorArray| where each vector is a direct sum of sub-vectors.
-
-    Given a list of equal length |VectorArrays| `blocks`, this |VectorArray|
-    represents the direct sums of the vectors contained in the arrays.
-    The associated |VectorSpace| is :class:`BlockVectorSpace`.
-
-    :class:`BlockVectorArray` can be used in conjunction with
-    :class:`~pymor.operators.block.BlockOperator`.
-    """
+class BlockVectorArrayImpl(VectorArrayImpl):
```

Comment:
Is this class referenced elsewhere in a docstring? Should it get one that maybe just refers to the `VectorArrayImpl`?